types/index.d.ts

```diff
@@ -141,16 +140,16 @@ declare namespace Realm {
         error?: ErrorCallback;
     }
 
-    // Note: This type does not work correctly without strictNullChecks enabled –
-    // a config of { flexible: true } will incorrectly have a type error
+    // Note: Since there really is no use case for {flexible: false}, we are
+    // hard setting this to only allowing "true".
```

Comment:
It's not a big deal but maybe we should explain the reason behind this

```suggestion
    // We only allow `flexible` to be `true` or `undefined` - `{ flexible: false
    // }` is not allowed. This is because TypeScript cannot discriminate that
    // type correctly with `strictNullChecks` disabled, and there's no real use
    // case for `{ flexible: false }`.
```